packages/react-router/lib/types/route-module.ts

```diff
@@ -148,7 +148,9 @@ type ClientDataFunctionArgs<T extends RouteInfo> = {
    * from your route middlewares.  You may pass in initial context values in your
    * `<HydratedRouter unstable_getContext>` prop
    */
-  context: unstable_RouterContextProvider;
+  context: MiddlewareEnabled extends true
+    ? unstable_RouterContextProvider
+    : undefined;
```

Comment:
See my comment in the linked issue but this is correct - could we instead update the JSDoc above?

```
  /**
   * This is a `RouterContextProvider` instance generated from in to your 
   * routers `unstable_getContext` function and/or populated from route 
   * middleware functions.
   */
```